ecs/pod_creator.go

```diff
@@ -27,11 +31,227 @@ func NewBasicECSPodCreator(c cocoa.ECSClient, v cocoa.Vault) (*BasicECSPodCreato
+	placementStrat := ecs.PlacementStrategy{}
+	placementStrat.SetType(string(*strategy)).SetField(*param)
+	return []*ecs.PlacementStrategy{&placementStrat}
+}
+
+// exportEnvVars translates the environment variables into ECS environment variables and secrets.
+func (m *BasicECSPodCreator) exportEnvVars(ctx context.Context, variables []cocoa.EnvironmentVariable) ([]*ecs.KeyValuePair, []cocoa.PodSecret, []cocoa.NamedSecret, error) {
+	keyValuePairs := []*ecs.KeyValuePair{}
+	secrets := []cocoa.PodSecret{}
+	createSecrets := []cocoa.NamedSecret{}
+
+	for _, variable := range variables {
+		if variable.SecretOpts == nil {
+			keyValue := ecs.KeyValuePair{}
+			keyValue.SetName(*variable.Name).SetValue(*variable.Value)
+			keyValuePairs = append(keyValuePairs, &keyValue)
+		} else {
+			if !*variable.SecretOpts.Exists {
+				createSecrets = append(createSecrets, variable.SecretOpts.PodSecret.NamedSecret)
+			}
+			secrets = append(secrets, variable.SecretOpts.PodSecret)
+		}
+	}
+
+	return keyValuePairs, secrets, createSecrets, nil
+}
+
+// exportSecrets extracts all secrets and environment variables from ECS creation options
+func (m *BasicECSPodCreator) exportSecrets(ctx context.Context, merged cocoa.ECSPodCreationOptions) ([]*ecs.KeyValuePair, []cocoa.PodSecret, []cocoa.NamedSecret, error) {
+	allEnvVars := []*ecs.KeyValuePair{}
+	allSecrets := []cocoa.PodSecret{}
+	allSecretsToCreate := []cocoa.NamedSecret{}
+
+	for _, def := range merged.ContainerDefinitions {
+		envVars, secrets, secretsToCreate, err := m.exportEnvVars(ctx, def.EnvVars)
+		if err != nil {
+			return nil, nil, nil, err
+		}
+		allEnvVars = append(allEnvVars, envVars...)
+		allSecrets = append(allSecrets, secrets...)
+		allSecretsToCreate = append(allSecretsToCreate, secretsToCreate...)
+	}
+
+	return allEnvVars, allSecrets, allSecretsToCreate, nil
+}
+
+// createSecrets creates secrets that do not already exist
+func (m *BasicECSPodCreator) createSecrets(ctx context.Context, secrets []cocoa.NamedSecret) error {
+	for _, secret := range secrets {
+		_, err := m.vault.CreateSecret(ctx, secret)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// translateSecrets translates a PodSecret to an ECS Secret
+func translateSecrets(ctx context.Context, secrets []cocoa.PodSecret) []*ecs.Secret {
+	ecsSecrets := []*ecs.Secret{}
+
+	for _, secret := range secrets {
+		ecsSecret := ecs.Secret{}
+		ecsSecret.SetName(*secret.Name)
+		ecsSecret.SetValueFrom(*secret.Value)
+		ecsSecrets = append(ecsSecrets, &ecsSecret)
+	}
+
+	return ecsSecrets
+}
+
+// exportPodCreationOptions converts options to create a pod into its equivalent ECS task definition.
+func (m *BasicECSPodCreator) exportPodCreationOptions(ctx context.Context, merged cocoa.ECSPodCreationOptions, envVars []*ecs.KeyValuePair, secrets []cocoa.PodSecret) (*ecs.RegisterTaskDefinitionInput, error) {
+	var containerDefs []*ecs.ContainerDefinition
+
+	for _, def := range merged.ContainerDefinitions {
+		if def.CPU == nil {
```

Comment:
This is not a required parameter. You can just do something like this:
```go
containerDef.SetCPU(int64(utility.FromIntPtr(def.CPU))
```

Using `utility` conversion functions is preferable to just dereferencing pointers since they'll never panic. If the pointer is nil, it'll return the value's zero type (in this case `int64(0)`).